django/contrib/gis/db/backends/base/operations.py

```diff
@@ -104,7 +104,12 @@ def get_geom_placeholder(self, f, value, compiler):
 
     def check_aggregate_support(self, aggregate):
         if aggregate.contains_aggregate == 'gis':
```

Comment:
Mostly looks fine to me. If it gets merged, it'll make what I plan to do a lot easier, which is create a list of disallowed expressions, of which aggregates will also be included.

Removal of the old gis.sql.aggregates.GeoAggregate is probably the wrong move. The regular aggregates.Aggregate base class has code in it to support old-style custom aggregates (see _patch_aggregate method), but I've not tested it with GeoAggregates. We can remove this in 2.0 I think.